discovery/watcher.go

```diff
@@ -193,13 +195,28 @@ func (w *Watcher) currentState() State {
 // Stop stops the Watcher after Run is called.
 // This cancels the Watcher's internal context.
 func (w *Watcher) Stop() {
+	w.log.Debug("Watcher.Stop")
+	// If applicable, attempt to log out. This must be done prior to the
+	// connection being closed. We ignore errors since we must continue to shut
+	// down.
+	//
+	// Do not use w.ctx which is likely already cancelled at this point.
+	if w.acls != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
```

Comment:
I changed this to not use `w.ctx` per the issue Iryna ran into: https://github.com/hashicorp/consul-dataplane/pull/7/files#r973220663 where `w.ctx` is probably already cancelled.

For now, I have it timeout if the Login takes more than 5 seconds but I'm wondering if, instead, `Stop` should accept a context.